resources/keycloak/custom-user-provider/src/main/java/com/cvmanager/auth/provider/user/CustomUserStorageProvider.java

```diff
@@ -0,0 +1,324 @@
+            st.execute();
+            ResultSet rs = st.getResultSet();
+            List<UserModel> users = new ArrayList<>();
+            while (rs.next()) {
+                users.add(new UserAdapter(ksession, realm, model, UserObject.fromJoinedResultSet(rs)));
+            }
+            return users.stream();
+        } catch (SQLException ex) {
+            throw new RuntimeException("Database error:" + ex.getMessage(), ex);
+        }
+    }
+
+    @Override
+    public Stream<UserModel> searchForUserStream(RealmModel realm, String search, Integer firstResult,
+            Integer maxResults) {
+        log.debug("searchForUserStream: realm={}", realm.getName());
+
+        try (Connection c = DbUtil.getConnection(this.model)) {
+            PreparedStatement st = this.getBaseUserQuery(c,
+                    search.equals("*") ? "" : String.format("WHERE email like %s", "%" + search + "%"),
+                    String.format("order by email limit %s offset %s", maxResults, firstResult));
+            log.debug("searchForUserStream: st={}", st);
+            st.execute();
+            ResultSet rs = st.getResultSet();
+            List<UserModel> users = new ArrayList<>();
+            while (rs.next()) {
+                users.add(new UserAdapter(ksession, realm, model, UserObject.fromJoinedResultSet(rs)));
+            }
+            return users.stream();
+        } catch (SQLException ex) {
+            throw new RuntimeException("Database error:" + ex.getMessage(), ex);
+        }
+    }
+
+    @Override
+    public Stream<UserModel> searchForUserStream(RealmModel realm, Map<String, String> params, Integer firstResult,
+            Integer maxResults) {
+        return getGroupMembersStream(realm, null, firstResult, maxResults);
+    }
+
+    @Override
+    public Stream<UserModel> searchForUserByUserAttributeStream(RealmModel realm, String attrName, String attrValue) {
+        // TODO: Cast variables of certain types
+        log.debug("searchForUserByUserAttributeStream: realm={}, attrName={}, attrValue={}", realm.getName(),
+                attrName, attrValue);
+        try (Connection c = DbUtil.getConnection(this.model)) {
+            PreparedStatement st = this.getBaseUserQuery(c,
+                    String.format("WHERE %s = %s", attrName, attrValue),
+                    String.format("order by email"));
+            log.debug("searchForUserByUserAttributeStream: st={}", st);
+            ResultSet rs = st.executeQuery();
+            List<UserModel> users = new ArrayList<>();
+            while (rs.next()) {
+                users.add(new UserAdapter(ksession, realm, model, UserObject.fromJoinedResultSet(rs)));
+            }
+            return users.stream();
+        } catch (SQLException ex) {
+            log.error("Database error in searchForUserByUserAttributeStream:", ex);
+            throw new RuntimeException("Database error:" + ex.getMessage(), ex);
```

Comment:
question(non-blocking): to reduce duplication, should we consider creating something like a RuntimeStorageException that does what this line is doing? We have at least 8 occurrences of `RuntimeException("Database error:" + ex.getMessage(), ex)` in this file

Reply:
Added a RuntimeStorageException class to reduce code duplication